src/game/SpellEffects.cpp

```diff
@@ -7147,10 +7155,20 @@ void Spell::EffectSummonDeadPet(SpellEffectIndex /*eff_idx*/)
 {
     if (m_caster->GetTypeId() != TYPEID_PLAYER)
         return;
+
+    bool wasDespawned = false;
     Player* _player = (Player*)m_caster;
     Pet* pet = _player->GetPet();
     if (!pet)
-        return;
+    {
+        pet = new Pet;
+        if (!pet->LoadPetFromDB(_player, 0, 0, false, true))
```

Comment:
No because that would be against the flow of the code. We would be sending result of spell twice, once generically in Spell::cast and once in the handler. To catch an error here its too late. LoadFromDB will throw an error to the console, which is enough.